compiler/p/codegen/FPTreeEvaluator.cpp

```diff
@@ -393,81 +357,15 @@ TR::Register *OMR::Power::TreeEvaluator::fconstEvaluator(TR::Node *node, TR::Cod
 
 TR::Register *OMR::Power::TreeEvaluator::dconstEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   double            value = node->getDouble();
-   TR::InstOpCode::Mnemonic     opcode = TR::InstOpCode::lfd;
-   bool              splats = false;
+   double value = node->getDouble();
    TR::Compilation *comp = cg->comp();
 
-   if (node->getOpCodeValue() == TR::vsplats)
-      {
-      splats = true;
-      }
-
-   if (splats)
-      {
-      value = node->getFirstChild()->getDouble();
-      opcode = TR::InstOpCode::lxvdsx;
-      }
-
-
-   if (!splats)
-      {
-      if (value == ((double) ((float) value)))
-         return fconstHandler(node, cg, (float) value);
-      }
-
-   TR::Register *trgRegister;
-
-   if (splats)
-      trgRegister = cg->allocateRegister(TR_VSX_VECTOR);
-   else
-      trgRegister = cg->allocateRegister(TR_FPR);
-
-   TR::Register       *srcRegister, *tempReg=NULL;
-   TR::Instruction *q[4];
-   int32_t            offset;
-
-   if (cg->comp()->target().is64Bit())
-      {
-      offset = cg->findOrCreateFloatConstant(&value, TR::Double, NULL, NULL, NULL, NULL);
-      if (offset != PTOC_FULL_INDEX)
-         {
-         if (offset<LOWER_IMMED || offset>UPPER_IMMED)
-            {
-            srcRegister = cg->allocateRegister();
+   if (value == ((double) ((float) value)))
```

Comment:
Trying this optimization looks unwieldy and possibly incorrect. Constants from longBits2D, or -0 vs. +0, for example.  We typically used integer bit-pattern for float equality comparison by the way.

Reply:
This was already there and was just moved by refactoring, but I agree that this is a potentially dangerous optimization to be trying to do. I've gone ahead and removed it.